lib/kino/data_table.ex

```diff
@@ -49,18 +49,19 @@ defmodule Kino.DataTable do
 
     name = Keyword.get(opts, :name, "Data")
     sorting_enabled = Keyword.get(opts, :sorting_enabled, true)
+    keys = opts[:keys]
 
-    {data_rows, data_columns} =
-      if keys = opts[:keys] do
-        {rows, %{columns: columns}} = Table.to_rows_with_info(tabular, only: keys)
-        nonexistent = keys -- columns
-        {rows, keys -- nonexistent}
+    {data_rows, data_columns, count} =
+      if keys do
+        {rows, info} = Table.to_rows_with_info(tabular, only: keys)
+        nonexistent = keys -- info.columns
+        {rows, keys -- nonexistent, info.count}
       else
-        {rows, %{columns: columns}} = Table.to_rows_with_info(tabular)
-        {rows, columns}
+        {rows, info} = Table.to_rows_with_info(tabular)
+        {rows, info.columns, info.count}
       end
```

Comment:
@josevalim for the built-in list `Table.Reader` implementation, we can't special case by calling `init` and checking the enumerable, because the enumerable is actually a stream that normalizes maps/keyval lists into list of columns.

One way to solve this would be adding `count: length(...)` to the metadata, since the protocol implementation knows that all the data is in memory. The downside is that we need to traverse all rows, which we wanted to avoid initially.

Alternatively we could have a flag that indicates if the data is in memory or lazily loaded, and then we know if we can do `Enum.count` ourselves. Though this doesn't feel elegant to me.

Reply:
Right... that's why I think we should call `Table.Reader.init` directly and allow the result of `init` to be given to `to_rows` and friends.

Reply:
Yeah, but my point is that even calling `Table.Reader.init` we don't know if we can call `Enum.count`, because the enumerable is a stream, not a list.